File: src/tui/clipboard.ts
/**
 * Clipboard reading for TUI paste support.
 * Uses platform-native commands to read clipboard text.
 */

import { execa } from "execa";

export async function readClipboard(): Promise<string> {
  try {
    let result: { stdout: string };

    if (process.platform === "win32") {
      result = await execa("powershell", ["-NoProfile", "-Command", "Get-Clipboard"], {
        timeout: 3_000,
      });
    } else if (process.platform === "darwin") {
      result = await execa("pbpaste", [], {
        timeout: 3_000,
      });
    } else {
      // Linux: try xclip, then xsel
      try {
        result = await execa("xclip", ["-selection", "clipboard", "-o"], {
          timeout: 3_000,
        });
      } catch (err) {
        result = await execa("xsel", ["--clipboard", "--output"], {
          timeout: 3_000,
        });
      }
    }

    return (result.stdout || "").trim();
  } catch (error) {
    return "";
  }
}

File: internal/tui/clipboard.go
package tui

import (
	"context"
	"os/exec"
	"runtime"
	"strings"
	"time"
)

// readClipboard reads text from the system clipboard.
// Returns empty string if clipboard is empty or reading fails.
func readClipboard() string {
	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
	defer cancel()

	var cmd *exec.Cmd
	switch runtime.GOOS {
	case "windows":
		cmd = exec.CommandContext(ctx, "powershell", "-NoProfile", "-Command", "Get-Clipboard")
	case "darwin":
		cmd = exec.CommandContext(ctx, "pbpaste")
	default:
		// Linux: try xclip first, then xsel
		cmd = exec.CommandContext(ctx, "xclip", "-selection", "clipboard", "-o")
		out, err := cmd.Output()
		if err == nil {
			return strings.TrimSpace(string(out))
		}
		cmd = exec.CommandContext(ctx, "xsel", "--clipboard", "--output")
	}

	out, err := cmd.Output()
	if err != nil {
		return ""
	}
	return strings.TrimSpace(string(out))
}
